server/routes/profile.go

```diff
@@ -52,12 +52,19 @@ func CheckDatastore(w http.ResponseWriter, r *http.Request) {
 }
 
 func GetProfile(w http.ResponseWriter, r *http.Request) {
-	ctx := context.Background()
+	if r.Method != http.MethodGet {
+		log.Printf("Invalid request method")
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.Write([]byte("Invalid request method"))
+		return
+	}
+
+	ctx := r.Context()
 
 	dsClient, err := datastore.NewClient(ctx, config.Project())
 	if err != nil {
-		log.Println(err)
-		w.WriteHeader(http.StatusOK)
+		log.Printf("Cannot connect to DataStore: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
I'd respond with 502 Bad Gateway here. Actually, imo you shouldn't send a response body if you respond with 5xx error